sw/host/ot_certs/src/cwt.rs

```diff
@@ -0,0 +1,354 @@
+            VariableType::ByteArray{size} => 1 + 8 + size,
+            VariableType::Integer => 1 + 8,
+            VariableType::String{size} => 1 + 8 + size,
+        }
+    }
+}
+
+/// Declaration of a variable that can be filled into the template.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(tag = "type", rename_all = "kebab-case")]
+pub enum ConstantType {
+    /// Raw array of bytes.
+    ByteArray {
+        /// Big-endian hex-encoded bytes
+        #[serde(with = "hex::serde")]
+        value: Vec<u8>,
+    },
+    /// Integer.
+    Integer {
+        value: i64,
+    },
+    /// UTF-8 encoded String.
+    String {
+        value: String,
+    },
+}
+
+impl ConstantType {
+    fn get_max_size(&self) -> usize {
+        match self {
+            ConstantType::ByteArray{value} => 1 + 8 + value.len(),
+            ConstantType::Integer{..} => 1 + 8,
+            ConstantType::String{value} => 1 + 8 + value.len(),
+        }
+    }
+}
+
+/// Declaration of a variable that can be filled into the template.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+#[serde(untagged)]
+pub enum StructureType {
+    Item(String),
+    Map(IndexMap<String, StructureType>),
+    Array(Vec<StructureType>),
+}
+
+impl StructureType {
+    fn get_max_size(&self, variables: &IndexMap<String, VariableType>, constants: &IndexMap<String, ConstantType>) -> Result<usize> {
+        let lookup = |var_name| {
+            if let Some(var) = variables.get(var_name) {
+                return Ok(var.get_max_size())
+            }
+
+            if let Some(var) = constants.get(var_name) {
+                return Ok(var.get_max_size())
+            }
+
+            return Err(anyhow!("Cannot find {} in variables or constants", var_name));
+        };
+
+        match self {
+            StructureType::Item(var_name) => lookup(var_name),
+            StructureType::Map(map) => {
+                let mut size: usize = 1 + 8;
+                for (k, v) in map.iter() {
+                    size += lookup(k)?;
+                    size += v.get_max_size(variables, constants)?;
+                }
+
+                Ok(size)
+            },
+            StructureType::Array(arr) => {
+                let mut size: usize = 1 + 8;
+                for v in arr.iter() {
+                    size += v.get_max_size(variables, constants)?;
+                }
+
+                Ok(size)
+            },
+        }
+    }
+}
+
+pub fn load_cwt_template(path: &PathBuf) -> Result<CwtTemplate> {
+    // Load template.
+    let template_content = fs::read_to_string(path)
+        .with_context(|| format!("Could not load the template file {}", path.display()))?;
+
+    CwtTemplate::from_hjson_str(&template_content)
+}
+
+impl CwtTemplate {
+    pub fn from_hjson_str(content: &str) -> Result<CwtTemplate> {
+        Ok(deser_hjson::from_str(content)?)
+    }
+}
+
+fn generate_c_var(template: &CwtTemplate, indent: &str) -> Result<String> {
```

Comment:
maybe document what this generates exactly. I think these are the variable definitions in the structure that holds all variables?